### PR TITLE
XRDDEV-864 delete client

### PR DESCRIPTION
**Pull request by Riippi**
Delete client
https://jira.niis.org/browse/XRDDEV-864

Also new unit tests for two methods in helpers

### CODE REVIEW

File: src/proxy-ui-api/frontend/src/views/Clients/Subsystem.vue [hunk 1]
```diff
@@ -66,6 +91,25 @@ export default Vue.extend({
...
+      }
+
+      if (
+        this.client.status === 'REGISTERED' ||
+        this.client.status === 'REGISTRATION_IN_PROGRESS'
+      ) {
+        return false;
+      }
+
+      if (this.$store.getters.hasPermission(Permissions.SEND_CLIENT_DEL_REQ)) {
```
Comment (TJaakkola) [inline src/proxy-ui-api/frontend/src/views/Clients/Subsystem.vue:106]: `return this.$store.getters.hasPermission(Permissions.SEND_CLIENT_DEL_REQ)`

File: src/proxy-ui-api/frontend/src/views/Clients/Subsystem.vue [hunk 2]
```diff
@@ -66,6 +91,25 @@ export default Vue.extend({
...
+      if (!this.client) {
+        return false;
+      }
+
+      if (
+        this.client.status === 'REGISTERED' ||
+        this.client.status === 'REGISTRATION_IN_PROGRESS'
+      ) {
+        return false;
+      }
```
Comment (TJaakkola) [inline src/proxy-ui-api/frontend/src/views/Clients/Subsystem.vue:104]: maybe just 
```
if (!this.client || this.client.status === 'REGISTERED' || this.client.status === 'REGISTRATION_IN_PROGRESS') {
  return false
}
```

File: src/proxy-ui-api/frontend/src/views/Clients/Subsystem.vue (hunk 1 above)
Comment (Riippi) [reply src/proxy-ui-api/frontend/src/views/Clients/Subsystem.vue:106]: Fixed

File: src/proxy-ui-api/frontend/src/views/Clients/Subsystem.vue (hunk 2 above)
Comment (Riippi) [reply src/proxy-ui-api/frontend/src/views/Clients/Subsystem.vue:104]: Fixed

**Pull request merged**